s3torchconnector/tst/unit/test_user_agent.py

```diff
@@ -0,0 +1,33 @@
+#  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#  // SPDX-License-Identifier: BSD
+
+from s3torchconnector._version import __version__
+from s3torchconnector._user_agent import UserAgent
+
+
+def test_default_user_agent():
+    user_agent = UserAgent()
+    assert user_agent.prefix == f"s3torchconnector/{__version__}"
+
+
+def test_user_agent_creation_with_empty_comments():
+    user_agent = UserAgent([])
+    assert user_agent.prefix == f"s3torchconnector/{__version__}"
+
+
+def test_user_agent_creation_with_empty_str_comments():
+    user_agent = UserAgent([""])
+    assert user_agent.prefix == f"s3torchconnector/{__version__}"
+
+
+def test_user_agent_creation_with_none_comments():
+    user_agent = UserAgent(None)
+    assert user_agent.prefix == f"s3torchconnector/{__version__}"
+
+
+def test_user_agent_creation_with_comments():
```

Comment:
Not blocking: Might want to consider a hypothesis test that asserts the prefix always starts with `s3torchconnector/{__version__}`, and always includes the comments given in the string